### ISSUE TITLE
yolo local

### ISSUE THREAD

**Issue by SongXueZhi** [SongXueZhi/auto-game-player#1]
import torch

# 切换到YOLOv5的目录
import os
os.chdir('yolov5')

# 导入YOLOv5模型
from models.yolo import Model

# 加载模型配置文件
config = 'models/yolov5s.yaml'  # 这里可以选择不同大小的YOLOv5模型（yolov5s, yolov5m, yolov5l, yolov5x）
model = Model(config)

# 加载权重文件
weights_path = '../weights/best.pt'  # 修改为你本地权重文件的路径
model.load_state_dict(torch.load(weights_path, map_location=torch.device('cpu'))['model'].float().state_dict())

# 将模型设置为评估模式
model.eval()

**Comment by SongXueZhi** [comment]
python train.py --img 1280 --epochs 100 --data data\\dnf.yaml --weights yolov5s.pt
我使用以上命令训练了微调模型，之后如何使用该模型

**Comment by SongXueZhi** [comment]
import torch
import win32api
import os
from models.yolo import Model
from PIL import Image

# 加载模型配置文件
config = 'models/yolov5s.yaml'  # 这里可以选择不同大小的YOLOv5模型（yolov5s, yolov5m, yolov5l, yolov5x）
model = Model(config)

# 加载权重文件
weights_path = 'D:\\dnf\\yolov5\\runs\\train\\exp7\\weights/best.pt'  # 修改为你本地权重文件的路径
model.load_state_dict(torch.load(weights_path, map_location=torch.device('cpu'))['model'].float().state_dict(), strict=False)

# 将模型设置为评估模式
model.eval()

with open('classes.txt', 'r') as f:
    classes = [line.strip() for line in f.readlines()]

hwnd = win32gui.FindWindow(None, "地下城与勇士：创新世纪")
# Bring the DNF game window to the foreground and set focus
win32gui.SetForegroundWindow(hwnd)
# Get the screen size
screen_width, screen_height = win32api.GetSystemMetrics(0), win32api.GetSystemMetrics(1)

# Move the game window to the top-left corner of the screen
win32gui.SetWindowPos(hwnd, 0, 0, 0, 0, 0, 0x0001)
# Get the game window position and size
left, top, right, bottom = win32gui.GetClientRect(hwnd)
width = right - left
height = bottom - top

# Loop to capture screenshots
for i in range(0,30):
    # Get the screen shot
    img = ImageGrab.grab(bbox=(left, top, right, bottom))

    # Convert the image to an OpenCV format
    img_np = np.array(img)
    img_cv = cv2.cvtColor(img_np, cv2.COLOR_BGR2RGB)
    img_tensor = torch.from_numpy(img_cv).permute(2, 0, 1).float().unsqueeze(0)

    # Use YOLOv5 to detect objects in the image
    results = model(img_tensor)

    # 获取结果中的检测框和类别
    detections = results[0]

**Comment by SongXueZhi** [comment]
import torch
import win32api
import win32gui
import os
from models.yolo import Model
from PIL import Image, ImageGrab
import torchvision.transforms as transforms
from utils.general import non_max_suppression, scale_coords

# ... （此处省略了前面的代码，包括模型实例化和权重加载）

# 定义图像预处理
img_transforms = transforms.Compose([
    transforms.Resize((1280, 1280)),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
])

# 存储检测结果的列表
detection_results = []

# 循环捕获屏幕截图并进行目标检测
for i in range(0, 30):
    # 获取屏幕截图
    img = ImageGrab.grab(bbox=(left, top, right, bottom))
    
    # 图像预处理
    input_image = img_transforms(img).unsqueeze(0)
    
    # 使用模型进行预测
    with torch.no_grad():
        output = model(input_image)
    
    # 解析输出结果
    conf_thres = 0.25  # 置信度阈值
    iou_thres = 0.45   # IOU阈值
    predictions = non_max_suppression(output, conf_thres, iou_thres)

    # 将预测框坐标缩放回原始图像大小
    for det in predictions:
        if len(det):
            det[:, :4] = scale_coords(input_image.shape[2:], det[:, :4], img.size).round()
    
    # 保存检测结果
    detection_results.append(predictions)

print(detection_results)

**Comment by SongXueZhi** [comment]
`def scale_coords(input_shape, coords, img_size):
    # 输入图像的高度和宽度
    input_height, input_width = input_shape
    img_height, img_width = img_size
    
    # 计算缩放比例
    scale_x = img_width / input_width
    scale_y = img_height / input_height

    # 缩放坐标
    coords[:, 0] *= scale_x  # x1
    coords[:, 1] *= scale_y  # y1
    coords[:, 2] *= scale_x  # x2
    coords[:, 3] *= scale_y  # y2
    
    return coords
`

**Comment by SongXueZhi** [comment]
```
    # 绘制边界框和类别标签
    draw = ImageDraw.Draw(img)
    font = ImageFont.truetype("arial.ttf", 20)  # 字体文件路径和字体大小
    for prediction in predictions:
        for box in prediction:
            x1, y1, x2, y2, conf, cls = box
            draw.rectangle([(x1, y1), (x2, y2)], outline="red", width=2)
            draw.text((x1, y1 - 20), f"{classes[int(cls)]} {conf:.2f}", fill="red", font=font)

    # 保存带有检测结果的图像
    img.save(f"result_{i}.png")

print(detection_results)
```
